### PR TITLE
Lifestudy preview

### PR DESCRIPTION
**Pull request by samvictor232**
Assign life study discipline based on attendance

### CODE REVIEW

File: ap/accounts/models.py [hunk 1]
```diff
@@ -352,62 +354,134 @@ def events_in_week_list(self, weeks):
...
+      e = str(datetime.combine(slip['rolls__date'], slip['rolls__event__end'])).replace(' ', 'T')
+      return (s, e)
+
+    group_slips = GroupSlip.objects.filter(trainees=self, status='A')
+
+    rolls = self.current_rolls.exclude(status='P') #exclude all present rolls
+    if self.self_attendance:
+      rolls = rolls.filter(submitted_by=self)
+    else:
+      rolls = rolls.exclude(submitted_by=self)
```
Comment (yipingbenjaminliu) [inline ap/accounts/models.py]: ADD a TODO that it doesn't cover trainees who are also a team monitor

File: ap/accounts/models.py [hunk 2]
```diff
@@ -352,62 +354,134 @@ def events_in_week_list(self, weeks):
...
+    else:
+      rolls = rolls.exclude(submitted_by=self)    
+    rolls = rolls.order_by('event__id', 'date').distinct('event__id', 'date')  # may not need to order
+    week_list = list(range(20))
+
+    if period is not None:  # works without period, but makes calculate_summary really slow
+      p = Period(Term.current_term())
+      start_date = p.start(period)
+      end_date = p.end(period)
+      # TODO: Works sometimes.
```
Comment (yipingbenjaminliu) [inline ap/accounts/models.py]: what other times?

File: ap/accounts/models.py [hunk 3]
```diff
@@ -544,3 +618,17 @@ class Statistics(models.Model):
   latest_ls_chpt = models.CharField(max_length=400, null=True, blank=True)
 
   settings = JSONField(default=default_settings())
+
+
+@timeit
```
Comment (yipingbenjaminliu) [inline ap/accounts/models.py]: do we need this stuff?

File: ap/houses/models.py [hunk 4]
```diff
@@ -106,7 +106,7 @@ class Room(models.Model):
   # refers to number of beds
   capacity = models.SmallIntegerField(default=0)  # 0 if room is not a bedroom
 
-  house = models.ForeignKey(House, on_delete=models.SET_NULL, null=True)
```
Comment (yipingbenjaminliu) [inline ap/houses/models.py]: what?

File: ap/leaveslips/models.py [hunk 5]
```diff
@@ -183,6 +183,8 @@ def periods(self):
   def events(self):
     evs = []
     for roll in self.rolls.all():
+      if not roll.event:
```
Comment (yipingbenjaminliu) [inline ap/leaveslips/models.py]: what is this

File: ap/lifestudies/views.py [hunk 6]
```diff
@@ -1,31 +1,44 @@
 import datetime
 import logging
+import json
```
Comment (yipingbenjaminliu) [inline ap/lifestudies/views.py]: clean up imports, not all are used

File: ap/lifestudies/views.py [hunk 7]
```diff
@@ -307,10 +357,30 @@ def get_context_data(self, **kwargs):
     context['date_today'] = datetime.date.today()
     return context
 
-
+'''
```
Comment (yipingbenjaminliu) [inline ap/lifestudies/views.py]: do we need this?

File: ap/schedules/management/commands/populate_events.py [hunk 8]
```diff
@@ -367,7 +368,43 @@ def _create_events(self):
     for i in [1, 3]:
       e = Event(name=name, code=code, type=type, start=start, end=end, weekday=i, monitor='AM', description=description)
       e.save()
-
+    '''
```
Comment (yipingbenjaminliu) [inline ap/schedules/management/commands/populate_events.py]: please get rid of this.

File: ap/xb_application/views.py [hunk 9]
```diff
@@ -37,9 +37,6 @@ def form_valid(self, form):
 
   def get_context_data(self, **kwargs):
     ctx = super(XBApplicationView, self).get_context_data(**kwargs)
-    self.object = self.get_object()
```
Comment (yipingbenjaminliu) [inline ap/xb_application/views.py]: why is this being touched here

File: libraries/bootflat-ftta/bootflat/scss/bootflat/_table_datatable.scss [hunk 10]
```diff
@@ -8,6 +8,9 @@
   #content form {
     margin-top: 7px;
   }
+  table.dataTable tbody > tr.selected {
```
Comment (yipingbenjaminliu) [inline libraries/bootflat-ftta/bootflat/scss/bootflat/_table_datatable.scss]: did we need to touch this?

File: ap/accounts/models.py (hunk 2 above)
Comment (samvictor232) [reply ap/accounts/models.py]: @iamsitting

File: ap/accounts/models.py (hunk 3 above)
Comment (samvictor232) [reply ap/accounts/models.py]: @iamsitting

File: ap/houses/models.py (hunk 4 above)
Comment (samvictor232) [reply ap/houses/models.py]: @billyeh

File: ap/leaveslips/models.py (hunk 5 above)
Comment (samvictor232) [reply ap/leaveslips/models.py]: @billyeh

File: ap/lifestudies/views.py (hunk 7 above)
Comment (samvictor232) [reply ap/lifestudies/views.py]: Removed

File: libraries/bootflat-ftta/bootflat/scss/bootflat/_table_datatable.scss (hunk 10 above)
Comment (samvictor232) [reply libraries/bootflat-ftta/bootflat/scss/bootflat/_table_datatable.scss]: @billyeh

File: ap/accounts/models.py (hunk 3 above)
Comment (iamsitting) [reply ap/accounts/models.py]: It's a test line... no.

File: ap/accounts/models.py (hunk 2 above)
Comment (iamsitting) [reply ap/accounts/models.py]: Red comment 4 lines above: "# works without period, but makes calculate_summary really slow"

**Comment by iamsitting** [comment]
For increasing/decreasing discipline on detail-discipline, let's do a numeric input. The numeric input should show the current discipline quantity and the TA should be to increase or decrease the quantity by inputting a number. Make sure that there won't be a negative "due" count.

**Comment by samvictor232** [comment]
@iamsitting 
Will make a new branch to make those changes.

**Pull request merged**